### ISSUE TITLE
fillgraph bug

### ISSUE THREAD

**Issue by viktorvi** [danvk/dygraphs#558]
![1](https://cloud.githubusercontent.com/assets/10329678/6542515/d7a6d0f6-c4fa-11e4-90ba-d3a96570a589.png)

When I enable fillgraph option then dygraph fill my graph incorrectly. Here is an example:

http://jsfiddle.net/3tsuw7j1/

When I zoom into the chart then display problem disappear.

**Comment by danvk** [comment]
Here's a version of that fiddle which renders in chrome: http://jsfiddle.net/3tsuw7j1/1/

**Comment by viktorvi** [comment]
http://jsfiddle.net/3tsuw7j1/1/ include the same display bug.

**Comment by danvk** [comment]
This is about as simple as I can get this repro: http://jsfiddle.net/3tsuw7j1/4/

The fix from #555 doesn't entirely work. It eliminates the second artifact but not the first.

**Comment by danvk** [comment]
The reason I couldn't get it any simpler is [this check][1]:

    if (points.length > 2 * g.width_) {
      ctx = DygraphCanvasRenderer._fastCanvasProxy(ctx);
    }

So with a 480px wide chart, you can't repro with under 960 points. Disabling that check, I can reproduce with just a few points.

```
["X,Y",
 "0, 0",
 "156145, 46",
 "156192, 101",
 "156431, 38",
 "160967, 0",
"1393728, 0"
].join('\n')
```

[1]: https://github.com/danvk/dygraphs/blob/097d2e0518ffb824d771cd83908ef84c5dd6b95e/src/dygraph-canvas.js#L760
